src/main/java/ee/carlrobert/codegpt/credentials/manager/AzureCredentialsManager.java

```diff
@@ -21,6 +21,7 @@ public AzureCredentialsManager() {
 
   @Override
   public void apply(AzureCredentials credentials) {
+    this.credentials = credentials;
```

Comment:
> > With "recognized" you mean your old apiKey was not loaded properly? I just realized I changed the CredentialAttributes by mistake for OpenAI + Azure, I changed it back to the same String key as before.
> 
> Yep, I've just pulled in the latest changes and am still experiencing the same issues. The Azure credentials appear to be affected as well. After building and installing the new version of the plugin on my IDE, the keys are not being loaded correctly, despite being stored in my keychain.

@carlrobertoh  I see I forgot to set the credentials not only in the PasswordSafe but also in the `CredentialsManager` too, but I dont think that has anything to do with your problem. The only lines where the `CredentialsUtil` is used is in the `CredentialsManager`s `apply` method and their constructor. I cannot reproduce your problem, since I didn't have any apiKey setup in my actual IntelliJ, only in the development build.

I will try to write some unit tests for the `CredentialsManager` and Settings

Reply:
I added the `ServiceSettingsTest` which tests every service's `Settings` and `CredentialManager` (`apply`, `isModified`)
https://github.com/carlrobertoh/CodeGPT/blob/392cceea2f1dfb40084bb547d6f707023f4ba052/src/test/java/ee/carlrobert/codegpt/settings/state/ServiceSettingsTest.java#L43

Reply:
Having trouble with the `CredentialsManagerTest` and `ServiceSettingsStateTest` being run in parallel, I think because they both set `ApplicationManager.setApplication`...

Reply:
I fixed the Tests by converting the `CredentialsUtil` to a service as well. I added constructors to manually pass in `CredentialsManagers` to the `Settings` and the `PasswordSafe` to the `CredentialsUtil`. This allows me to test the `Settings` in a plain Unit Test, without using `ApplicationManager` or `BasePlatformTestCase`